### ISSUE TITLE
Wrong type of super().__init__ when super() result type is Unknown

### ISSUE THREAD

**Issue by NateScarlet** [microsoft/pyright#1638]
**Describe the bug**
`super().__init__` type is `() -> None` when super class is unknown

**To Reproduce**
Write code provided below in vscode

**Expected behavior**
`__init__` should be `Unknown` type

**Screenshots or Code**.
```python
from some_module import ClassUnknown

class Class1(ClassUnknown):
    def __init__(self, x: int):
        super(Class1, self).__init__(x)

class Class2(ClassUnknown):
    def __init__(self, x: int):
        super_class = super(Class2, self)
        super_class.__init__(x)
```

![image](https://user-images.githubusercontent.com/21083014/111421839-a0bb4080-8728-11eb-829d-cd77ac5d1011.png)

**VS Code extension or command-line**
Pylance v2021.3.1

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

@jakebailey, this was due to a regression you recently introduced. I've added a test case to prevent future similar regressions. https://github.com/microsoft/pyright/commit/8a23fa119110b45dd9c934db74e6f7d3ff29d91e

**Comment by jakebailey** [comment]
Hm, was a part of Bill's docstring inheritance change that changed things to an interator so we could continue walking up. Thanks for the test.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.123, which I just published. It will also be included in the next release of pylance.
